connectors/sources/servicenow.py

```diff
@@ -0,0 +1,346 @@
+        """Get the default configuration for ServiceNow.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "url": {
+                "label": "Service URL",
+                "order": 1,
+                "type": "str",
+                "value": "http://127.0.0.1:9318",
+            },
+            "username": {
+                "label": "Username",
+                "order": 2,
+                "type": "str",
+                "value": "admin",
+            },
+            "password": {
+                "label": "Password",
+                "sensitive": True,
+                "order": 3,
+                "type": "str",
+                "value": "changeme",
+            },
+            "services": {
+                "label": "List of services",
+                "order": 4,
+                "type": "list",
+                "value": "*",
+            },
+            "retry_count": {
+                "display_value": RETRIES,
+                "display": "numeric",
+                "label": "Maximum retries per request",
+                "order": 5,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": RETRIES,
+            },
+            "concurrent_downloads": {
+                "default_value": MAX_CONCURRENCY_SUPPORT,
+                "display": "numeric",
+                "label": "Maximum concurrent downloads",
+                "order": 6,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": MAX_CONCURRENCY_SUPPORT,
+            },
+        }
+
+    async def _remote_validation(self):
+        """Validate configured services
+
+        Raises:
+            ConfigurableFieldValueError: Unavaliable services error.
+        """
+
+        if self.servicenow_client.services != ["*"]:
+            (
+                service_names,
+                invalid_services,
+            ) = await self.servicenow_client.filter_services()
+            if invalid_services:
+                raise ConfigurableFieldValueError(
+                    f"Services '{', '.join(invalid_services)}' are not available. Available services are: '{', '.join(service_names)}'"
+                )
+
+    async def validate_config(self):
+        """Validates whether user input is empty or not for configuration fields
+        Also validate, if user configured services are available in ServiceNow."""
+
+        logger.info("Validating ServiceNow Configuration")
+        self.configuration.check_valid()
+        await self._remote_validation()
+
+    async def close(self):
+        """Closes servicenow client."""
+
+        await self.servicenow_client.close_session()
+
+    async def ping(self):
+        """Verify the connection with ServiceNow."""
+
+        try:
+            logger.info("Pinging ServiceNow instance")
```

Comment:
```suggestion
```

This is also redundant IMO as `sync_job_runner` logs something similar before creating a sync.